### DISCUSSION TITLE
The routeParams object return error

### DISCUSSION THREAD

**Discussion by haoynmail** [vikejs/vike#929] [Discussion] [answered]
import { PageContextBuiltIn } from 'vite-plugin-ssr/types'

export default (pageContext: PageContextBuiltIn) => {
    const { urlPathname, urlParsed } = pageContext
    if (urlPathname === '/') {
        const { start } = urlParsed.search
        return { routeParams: { start } }
    }

    return false
}

**Comment by brillout** [comment]
What's the error?

**Comment by haoynmail** [comment]
I have a link component with query parameters, and I use urlParsed to obtain the query parameters. The components are as follows

import React from "react";
`export function Links() {

  return (

    <div className="links-wrap">

      <a href="/?start=500">

        start from <strong>500</strong>

      </a>

      <a href="/?start=1000">

        start from <strong>1000</strong>

      </a>

    </div>

  );

}`

My router function is as follows

import { PageContextBuiltIn } from 'vite-plugin-ssr/types'

export default (pageContext: PageContextBuiltIn) => {
    const { urlPathname, urlParsed } = pageContext
    if (urlPathname === '/') {
        const { start } = urlParsed.search
        return { routeParams: { start } }
    }

    return false
}

Error: [vite-plugin-ssr@0.4.123][Wrong Usage] The `routeParams` object returned by the Route Function /pages/index.page.route.ts should only hold string values.

**Reply by brillout** [reply, marked as answer]
https://brillout.github.io/rules/
